Lib/glyphsLib/builder/glyph.py

```diff
@@ -63,6 +68,47 @@ def to_ufo_glyph(self, ufo_glyph, layer, glyph):  # noqa: C901
                 layers = []
                 for layerId, colorId in layerMapping:
                     layers.append((glyph.layers[layerId], colorId))
+                self._color_palette_layers.append(((glyph, layer), layers))
+
+        if self.minimal:
+            # The other kind of color layers supports solid colors and
+            # gradients among other things, and we use it to build COLRv1
+            # table.
+            # For each color layer, we collect paths that has the same
+            # attributes. If there is only one group of such paths in a layer,
+            # the layer is used without changes, otherwise we make a clone of
+            # the layer for each group with only the paths in this group. We do
+            # this splitting because a COLRv1 layer can’t have multiple
+            # gradients or colors.
+            color_layers = [
+                l
+                for l in glyph.layers
+                if l.attributes.get("color")
+                and l.associatedMasterId == layer.associatedMasterId
+            ]
+            if color_layers:
+                import json
+
+                layers = []
+                for color_layer in color_layers:
+                    # XXX: What to do with components? Glyphs does not allow
+                    # setting attributes on them.
+                    assert not color_layer.components
```

Comment:
if these are valid and possible in source files, you should not just assert but raise a proper exception with error message

Reply:
They can occur in source files. That is a TODO item that I was planing to ask @schriftgestalt about.